letter-sender/src/main/java/com/christmas/letter/sender/controller/ChristmasLetterController.java

```diff
@@ -0,0 +1,44 @@
+package com.christmas.letter.sender.controller;
+
+import com.christmas.letter.sender.ChristmasLetter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+
+@RestController
+@RequestMapping("/letters")
+public class ChristmasLetterController {
+
+  private final SnsClient snsClient;
+
+  @Value("${sns.topic.arn}")
+  private String topicArn;
+
+  public ChristmasLetterController(SnsClient snsClient){
+    this.snsClient = snsClient;
+  }
+
+  @PostMapping("/send")
+  public String sendChristmasLetter(@RequestBody ChristmasLetter letter) {
+
+    String message = String.format("Email: %s\nName: %s\nWishes: %s\nLocation: %s",
```

Comment:
I wouldn't want my controller to directly invoke the sns client. Imagine we had more complex things to do here, they don't belong in the controller. Also, you can send the letter directly as a json representation, no need to reformat etc.

Reply:
You are right, I added a service layer